nautobot/extras/utils.py

```diff
@@ -207,6 +208,92 @@ def wrapper(model_class):
     return wrapper
 
 
+def update_registry_with_new_apps():
+    """
+    Updates the registry with new apps.
+
+    This function updates the registry model features for models.
+    """
+    from nautobot.extras.models import CustomField, Relationship, RelationshipAssociation, Role
+
+    lookup_confs = [
+        {
+            "model": CustomField,
+            "lookup_by": "field",
+            "field_names": ["_custom_field_data"],
+        },
+        {
+            "model": Relationship,
+            "lookup_by": "field",
+            "field_names": ["source_for_associations", "destination_for_associations"],
+            "extras": {"related_model": RelationshipAssociation},
+        },
+        {"model": Role, "lookup_by": "related_model", "related_model": Role},
+    ]
+
+    app_models = apps.get_models()
+    for lookup_conf in lookup_confs:
+        if lookup_conf["lookup_by"] == "field":
+            result = lookup_by_field(
+                app_models=app_models,
+                field_names=lookup_conf["field_names"],
+                extras=lookup_conf.get("extras"),
+            )
+        else:
+            result = lookup_by_related_model(app_models=app_models, related_model=lookup_conf["related_model"])
+        registry_items = {key: list(value) for key, value in result.items()}
+        feature_name = lookup_conf["model"]._meta.model_name
+        registry["model_features"][feature_name] = registry_items
+
+
+def lookup_by_field(app_models, field_names, extras=None):
+    """
+    Find all models that have fields with the specified names, and return them grouped by app.
+
+    Args:
+        app_models: A list of model classes to search through.
+        field_names: A list of names of fields to look for.
```

Comment:
If I understand this right, this will add the model to the result if *any* of these field names are valid? Should it instead only do so if *all* of the field names are valid?

Reply:
Still an open question, thanks! :-)

Reply:
Oooh Yes, if all field_name are present on the model